### PR TITLE
Add maxTrialDuration

### PR DESCRIPTION
**Pull request by acured**

### CODE REVIEW

File: ts/nni_manager/core/nnimanager.ts [hunk 1]
```diff
@@ -538,6 +542,22 @@ class NNIManager implements Manager {
         }
     }
 
+    private async stopTrialJobIfOverMaxDuration(): Promise<void> {
```
Comment (liuzhe-lz) [inline ts/nni_manager/core/nnimanager.ts:545]: I would suggest create a timer on each trial start.

Comment (acured) [reply ts/nni_manager/core/nnimanager.ts]: Changed.

File: ts/nni_manager/core/nnimanager.ts [hunk 2]
```diff
@@ -661,6 +676,7 @@ class NNIManager implements Manager {
                     this.currSubmittedTrialNum++;
                     this.log.info('submitTrialJob: form:', form);
                     const trialJobDetail: TrialJobDetail = await this.trainingService.submitTrialJob(form);
+                    setTimeout(async ()=> this.stopTrialJobIfOverMaxDurationTimer(trialJobDetail.id), 1000 * this.maxTrialDuration);
```
Comment (J-shang) [inline ts/nni_manager/core/nnimanager.ts:679]: do you test if this works?

Comment (acured) [reply ts/nni_manager/core/nnimanager.ts:679]: Yes, it is works, with "maxTrialDuration" set and not set.

**Review by J-shang — APPROVED** [review]

File: docs/en_US/Tutorial/ExperimentConfig.rst [hunk 3]
```diff
@@ -279,6 +280,15 @@ Optional. Integer between 1 and 99999. Default: 99999.
...
 Specifies the max number of trial jobs created by NNI, including succeeded and failed jobs.
 
+maxTrialDuration
+^^^^^^^^^^^^^^^^
+
+Optional. String. Default: 999d.
+
+**maxTrialDuration** specifies the max duration time of each trial job. The unit of the time is {**s**\ , **m**\ , **h**\ , **d**\ }, which means {*seconds*\ , *minutes*\ , *hours*\ , *days*\ }.
+
+Note: The maxTrialDuration spec set the time of each trial job. If current trial job reach the max duration time, this trial job will stop.
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/ExperimentConfig.rst:290]: you can remove "Note: ", because this is a normal description of "maxTrialDuration"

**Review by QuanluZhang — APPROVED** [review]

File: docs/en_US/Tutorial/ExperimentConfig.rst (hunk 3 above)
Comment (acured) [reply docs/en_US/Tutorial/ExperimentConfig.rst]: Removed

**Pull request merged**